executor/common_linux.h

```diff
@@ -1570,6 +1570,310 @@ static long syz_init_net_socket(volatile long domain, volatile long type, volati
 #endif
 #endif
 
+#if SYZ_EXECUTOR || SYZ_VHCI_INJECTION
+#include <pthread.h>
+#include <sys/epoll.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/uio.h>
+
+#define ACL_LINK 1
+#define SCAN_PAGE 2
+
+typedef struct {
+	uint8 b[6];
+} __attribute__((packed)) bdaddr_t;
+
+#define HCI_COMMAND_PKT 1
+struct hci_command_hdr {
+	uint16 opcode;
+	uint8 plen;
+} __attribute__((packed));
+
+#define HCI_EVENT_PKT 4
+struct hci_event_hdr {
+	uint8 evt;
+	uint8 plen;
+} __attribute__((packed));
+
+#define HCI_EV_CONN_COMPLETE 0x03
+struct hci_ev_conn_complete {
+	uint8 status;
+	uint16 handle;
+	bdaddr_t bdaddr;
+	uint8 link_type;
+	uint8 encr_mode;
+} __attribute__((packed));
+
+#define HCI_EV_CONN_REQUEST 0x04
+struct hci_ev_conn_request {
+	bdaddr_t bdaddr;
+	uint8 dev_class[3];
+	uint8 link_type;
+} __attribute__((packed));
+
+#define HCI_EV_REMOTE_FEATURES 0x0b
+struct hci_ev_remote_features {
+	uint8 status;
+	uint16 handle;
+	uint8 features[8];
+} __attribute__((packed));
+
+#define HCI_EV_CMD_COMPLETE 0x0e
+struct hci_ev_cmd_complete {
+	uint8 ncmd;
+	uint16 opcode;
+} __attribute__((packed));
+
+#define HCI_OP_READ_BUFFER_SIZE 0x1005
+struct hci_rp_read_buffer_size {
+	uint8 status;
+	uint16 acl_mtu;
+	uint8 sco_mtu;
+	uint16 acl_max_pkt;
+	uint16 sco_max_pkt;
+} __attribute__((packed));
+
+#define HCI_OP_READ_BD_ADDR 0x1009
+struct hci_rp_read_bd_addr {
+	uint8 status;
+	bdaddr_t bdaddr;
+} __attribute__((packed));
+
+struct hci_dev_stats {
+	uint32 err_rx;
+	uint32 err_tx;
+	uint32 cmd_tx;
+	uint32 evt_rx;
+	uint32 acl_tx;
+	uint32 acl_rx;
+	uint32 sco_tx;
+	uint32 sco_rx;
+	uint32 byte_rx;
+	uint32 byte_tx;
+};
+
+struct hci_dev_info {
+	uint16 dev_id;
+	char name[8];
+
+	bdaddr_t bdaddr;
+
+	uint32 flags;
+	uint8 type;
+
+	uint8 features[8];
+
+	uint32 pkt_type;
+	uint32 link_policy;
+	uint32 link_mode;
+
+	uint16 acl_mtu;
+	uint16 acl_pkts;
+	uint16 sco_mtu;
+	uint16 sco_pkts;
+
+	struct hci_dev_stats stat;
+};
+
+struct hci_dev_req {
+	uint16 dev_id;
+	uint32 dev_opt;
+};
+
+#define HCISETSCAN _IOW('H', 208, int)
+#define HCIGETDEVINFO _IOR('H', 211, int)
+
+#define BUF_SIZE 4096
+#define ACL_HANDLE 256
+#define HCI_DEV 0
+
+static int vhci_fd = -1;
+static int vhci_init = 0;
+
+static long syz_init_net_socket(volatile long domain, volatile long type, volatile long proto);
```

Comment:
This function is only available if:
```
#if SYZ_EXECUTOR || __NR_syz_init_net_socket
#if SYZ_EXECUTOR || SYZ_SANDBOX_NONE || SYZ_SANDBOX_SETUID || SYZ_SANDBOX_NAMESPACE || SYZ_SANDBOX_ANDROID
```

If the program does not use syz_init_net_socket, all C reproducers will fail to compile.

Reply:
Thanks!